modal/parallel_map.py

```diff
@@ -1226,6 +1355,16 @@ def handle_retry_response(self, input_jwts: list[str]):
             if ctx is not None:
                 ctx.handle_retry_response(input_jwt)
 
+    async def handle_check_inputs_response(self, response: list[tuple[int, bool]]):
+        for idx, lost in response:
+            ctx = self._item_context.get(idx, None)
+            if ctx is not None:
+                if lost:
+                    ctx.state = _MapItemState.WAITING_TO_RETRY
+                    retry_item = await ctx.create_map_start_or_continue_item(idx)
+                    delay_ms = ctx.retry_manager.get_delay_ms()
```

Comment:
In the control-plane version, the logic is to call `retry_manager.get_delay_ms()`, but then ignore the returned value, and set it to 0. This is because we want to increment the retry count (which happens when we call `get_delay_ms`), but we want to retry immediately. We should probably do the same thing here to keep parity.